homeassistant/auth/mfa_modules/totp.py

```diff
@@ -117,7 +117,8 @@ async def async_setup_flow(self, user_id: str) -> SetupFlow:
 
         Mfa module should extend SetupFlow
         """
-        user = await self.hass.auth.async_get_user(user_id)  # type: ignore
+        user = await self.hass.auth.async_get_user(user_id)
+        assert user is not None
```

Comment:
We should really get rid of this `assert`-ing.

It does nothing at runtime, it is merely to please MyPy. So we are adding in additional code for no reason, without function. It basically defeats the reason to type check in the first place.

Reply:
What we IMO really should be targeting is getting rid of the lots and lots of optionals the codebase is riddled with. It's a daunting task and one that can actually break code much more easier than just adding type hints and some workarounds needed in the meantime, and may require architectural changes. I don't think waiting for that to improve type hinting is a good choice.

But before (if ever) that happens, adding an assert like the above is better than adding a `type: ignore` -- it explicitly tells what we intend. The code after that will traceback anyway if the thing asserted to be not None for some reason is None. I don't think it defeats the reason to typecheck at all. Sprinkling `type: ignore`s on the other hand /instead kind of does.

Reply:
I agree on that, it at least carries it over the typing value now instead of completely ignoring it. I do think we can solve some of these cases nevertheless. My comment is in no way meant to be blocking on this PR. It is just... "Meh" it is that way right now for the same reasons as you stated.

Other than that, awesome job 👍

Reply:
Thanks for the clarification. And agreed very much on the "meh" :P